x-pack/dockerlogbeat/pipelinemanager/config.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transform/typeconv"
+)
+
+// ContainerOutputConfig has all the options we'll expect from --log-opts
+type ContainerOutputConfig struct {
+	Endpoint       []string `struct:"output.elasticsearch.hosts,omitempty"`
+	User           string   `struct:"output.elasticsearch.username,omitempty"`
+	Password       string   `struct:"output.elasticsearch.password,omitempty"`
+	Index          string   `struct:"output.elasticsearch.index,omitempty"`
+	Pipeline       string   `struct:"output.elasticsearch.pipeline,omitempty"`
+	APIKey         string   `struct:"output.elasticsearch.api_key,omitempty"`
+	Timeout        string   `struct:"output.elasticsearch.timeout,omitempty"`
+	BackoffInit    string   `struct:"output.elasticsearch.backoff.init,omitempty"`
+	BackoffMax     string   `struct:"output.elasticsearch.backoff.max,omitempty"`
+	CloudID        string   `struct:"cloud.id,omitempty"`
+	CloudAuth      string   `struct:"cloud.auth,omitempty"`
+	ProxyURL       string   `struct:"output.elasticsearch.proxy_url,omitempty"`
+	TemplateName   string   `struct:"setup.template.name,omitempty"`
+	TempatePattern string   `struct:"setup.template.pattern,omitempty"`
```

Comment:
Does the plugin install a template? If not let's remove the `TemplateName` and `TemplatePattern` from the list of configs. The settings in the `setup` namespace are really only for setup, that should be run before the output is started.